server/lib/filter.js

```diff
@@ -75,6 +76,29 @@ class FilterParser {
     }
   }
 
+
+  /**
+  *@description
+  * Allows to filter the elements relative to the results of a subRequest
+  * @param {String} principale    The element to be compared with the results of the sub-request
+  * @param {String} table         table to be used in subRequests
+  * @param {String} column        Is the equivalence between the table `tableString` and `table` that of the subRequests
+  * @param {String} filterKey     key attribute on filter object to be used in filter
+  */
+  subRequest(principale, table, column, filterKey) {
```

Comment:
This is a good idea and very well commented. 

However, calling:
```js
filters.subRequest('uuid', 'cash_item', 'invoice_uuid', 'cash_uuid');
```

creates the string:

```js
`cash.uuid IN (SELECT cash_item.invoice_uuid FROM cash_item WHERE cash_item.cash_uuid = HUID(?))`
```

Does this feature work? The cash UUID should never match the cash_item invoice uuid. 

This makes me think that the API for this function is too confusing, it uses the same parameter names as the other functions however they mean something different (tables and columns in a sub query). 

I would recommend that this behaviour is written using the `custom` method for now. 

Using this function and the function call, that would look like: 

```js
filters.custom('cash_uuid', 'cash.uuid IN (SELECT cash_item.cash_uuid FROM cash_item WHERE cash_item.invoice_uuid = HUID(?)');
```

I suggest this because the table, columns and filterKey provided here do not mean the same things as with the other functions.